sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/FeedResponseDiagnostics.java

```diff
@@ -66,49 +67,7 @@ public Map<String, QueryMetrics> getQueryMetricsMap() {
      */
     @Override
     public String toString() {
-        StringBuilder stringBuilder = new StringBuilder();
-        if (diagnosticsContext != null) {
-            stringBuilder.append(QUERY_PLAN + SPACE + QueryMetricsTextWriter.START_TIME_HEADER)
-                .append(EQUALS)
-                .append(QueryMetricsTextWriter.DATE_TIME_FORMATTER.format(diagnosticsContext.getStartTimeUTC()))
-                .append(System.lineSeparator());
-            stringBuilder.append(QUERY_PLAN + SPACE + QueryMetricsTextWriter.END_TIME_HEADER)
-                .append(EQUALS)
-                .append(QueryMetricsTextWriter.DATE_TIME_FORMATTER.format(diagnosticsContext.getEndTimeUTC()))
-                .append(System.lineSeparator());
-            if (diagnosticsContext.getStartTimeUTC() != null && diagnosticsContext.getEndTimeUTC() != null) {
-                stringBuilder.append(QUERY_PLAN + SPACE + QueryMetricsTextWriter.DURATION_HEADER)
-                    .append(EQUALS)
-                    .append(Duration.between(diagnosticsContext.getStartTimeUTC(),
-                        diagnosticsContext.getEndTimeUTC()).toMillis()).append(System.lineSeparator());
-                if (diagnosticsContext.getRequestTimeline() != null) {
-                    try {
-                        stringBuilder.append(QUERY_PLAN + SPACE + "RequestTimeline ")
-                            .append(EQUALS)
-                            .append(mapper.writeValueAsString(diagnosticsContext.getRequestTimeline()))
-                            .append(System.lineSeparator())
-                            .append(System.lineSeparator());
-                    } catch (JsonProcessingException e) {
-                        LOGGER.error("Error while parsing diagnostics ", e);
-                    }
-                }
-            }
-        }
-
-        if (queryMetricsMap != null && !queryMetricsMap.isEmpty()) {
-            queryMetricsMap.forEach((key, value) -> stringBuilder.append(key)
-                .append(EQUALS)
-                .append(value.toString())
-                .append(System.lineSeparator()));
-        }
-        try {
-            stringBuilder
-                .append(mapper.writeValueAsString(clientSideRequestStatistics));
-        } catch (JsonProcessingException e) {
-            LOGGER.error("Error while parsing diagnostics ", e);
-        }
-
-        return stringBuilder.toString();
+        return RntbdObjectMapper.toString(this);
```

Comment:
why are we using RntbdObjectMapper? We should re-use the one which is already defined in this class.

Reply:
RntbdObjectMapper also uses the object mapper only internally. But it abstracts a few exception handling. And I saw this being used in a similar case. But if it has to be used only within Rtnbd packages, I can change it.

Reply:
It also has additional configuration for object mapper without which using the plain objectMapper fails serialization. 
(https://github.com/Azure/azure-sdk-for-java/blob/b3679847c498627c8a7900e9f5d66e562e9f2c5a/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdObjectMapper.java#L38C5-L43C44)

Reply:
Got it, makes sense. 
@xinlian12  can you pls confirm if it is safe to use RntbdObjectMapper outside of rntbd package?

Reply:
I still think we should not use it outside of the rntbd package. I see we have one in this class, can't we customize it accordingly and use it here locally?

Reply:
Yes, we can do that.